hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -182,33 +182,41 @@ protected JpaDeploymentManagement(final EntityManager entityManager, final Actio
                 .map(entry -> DeploymentManagement.deploymentRequest(entry.getKey(), entry.getValue()).build())
                 .collect(Collectors.toList());
 
-        return assignDistributionSets(deploymentRequests, null, offlineDsAssignmentStrategy);
+        return assignDistributionSets(tenantAware.getCurrentUsername(), deploymentRequests, null,
+                offlineDsAssignmentStrategy);
     }
 
     @Override
     @Transactional(isolation = Isolation.READ_COMMITTED)
-    public List<DistributionSetAssignmentResult> assignDistributionSets(
+    public List<DistributionSetAssignmentResult> assignDistributionSets(final String triggeredBy,
```

Comment:
So I miss something or is this method never called from outside? ... just by the one below without the `triggeredBy ` in the signature?

Reply:
yes, you're right, will make it private

Reply:
If you mark it as private I assume you would also want to remove the @Transactional annotation because it will be not taken into consideration by Spring proxy. Nevertheless you can wrap it in explicit transaction if you think it is required here